src/plugins/kibana/server/lib/stitch_pattern_and_mappings.js

```diff
@@ -0,0 +1,12 @@
+const _ = require('lodash');
+
+module.exports = function stitchPatternAndMappings(pattern, mappings) {
```

Comment:
Can you describe what this should do and why. Unit tests with examples would help here.